app/controllers/local_transaction_controller.rb

```diff
@@ -34,19 +34,27 @@ def results
     @interaction_details = interaction_details
     @local_authority = LocalAuthorityPresenter.new(@interaction_details["local_authority"])
     @country_name = @local_authority.country_name
+    @devolved_administration_availability = devolved_administration_availability
     @service = LocalTransactionService.new(
-      @publication.title, lgsl, @country_name, @local_authority.url
+      @publication.title, lgsl, @country_name, @local_authority.url, @devolved_administration_availability
```

Comment:
Your reasoning about not wanting lots of arguments sounds good. I think rather than having a dedicated class for devolved_administration_availability it'd probably make sense to use the [LocalTransactionPresenter](https://github.com/alphagov/frontend/blob/master/app/presenters/local_transaction_presenter.rb) as a place to ask our devolved_administration questions. This is already set in this class and is @publication. We could then adapt the LocalTransactionService to accept the presenter as an argument rather than just the title - however I do question the need for LocalTransactionService in a later comment.